syntax = "proto3";

package yandex.cloud.audit.loadbalancer;

import "google/protobuf/timestamp.proto";
import "google/rpc/status.proto";
import "yandex/cloud/audit/common.proto";
import "yandex/cloud/validation.proto";
import "yandex/cloud/loadbalancer/v1/health_check.proto";
import "yandex/cloud/loadbalancer/v1/network_load_balancer.proto";

option go_package = "a.yandex-team.ru/cloud/bitbucket/public-api/yandex/cloud/audit/loadbalancer;loadbalancer";
option java_package = "yandex.cloud.api.audit.loadbalancer";
option java_outer_classname = "NetworkLoadBalancerEvents";

message CreateNetworkLoadBalancer {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string network_load_balancer_id = 1 [(required) = true];
        string network_load_balancer_name = 2 [(required) = true];
        string region_id = 3 [(required) = true];
        yandex.cloud.loadbalancer.v1.NetworkLoadBalancer.Type type = 4 [(required) = true];
        repeated yandex.cloud.loadbalancer.v1.Listener listeners = 5;
        repeated yandex.cloud.loadbalancer.v1.AttachedTargetGroup attached_target_groups = 6;
    }
}

message UpdateNetworkLoadBalancer {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string network_load_balancer_id = 1 [(required) = true];
        string network_load_balancer_name = 2 [(required) = true];
        string region_id = 3 [(required) = true];
        yandex.cloud.loadbalancer.v1.NetworkLoadBalancer.Type type = 4 [(required) = true];
        repeated yandex.cloud.loadbalancer.v1.Listener listeners = 5;
        repeated yandex.cloud.loadbalancer.v1.AttachedTargetGroup attached_target_groups = 6;
    }
}

message DeleteNetworkLoadBalancer {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string network_load_balancer_id = 1 [(required) = true];
        string network_load_balancer_name = 2 [(required) = true];
    }
}

message StartNetworkLoadBalancer {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string network_load_balancer_id = 1 [(required) = true];
        string network_load_balancer_name = 2 [(required) = true];
    }
}

message StopNetworkLoadBalancer {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string network_load_balancer_id = 1 [(required) = true];
        string network_load_balancer_name = 2 [(required) = true];
    }
}

message AttachNetworkLoadBalancerTargetGroup {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string network_load_balancer_id = 1 [(required) = true];
        string network_load_balancer_name = 2 [(required) = true];
        string target_group_id = 3 [(required) = true];
        string target_group_name = 4 [(required) = true];
        repeated yandex.cloud.loadbalancer.v1.HealthCheck health_checks = 5;
    }
}

message DetachNetworkLoadBalancerTargetGroup {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string network_load_balancer_id = 1 [(required) = true];
        string network_load_balancer_name = 2 [(required) = true];
        string target_group_id = 3 [(required) = true];
        string target_group_name = 4 [(required) = true];
        repeated yandex.cloud.loadbalancer.v1.HealthCheck health_checks = 5;
    }
}

message AddNetworkLoadBalancerListener {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string network_load_balancer_id = 1 [(required) = true];
        string network_load_balancer_name = 2 [(required) = true];
        yandex.cloud.loadbalancer.v1.Listener listener = 3 [(required) = true];
    }
}

message RemoveNetworkLoadBalancerListener {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string network_load_balancer_id = 1 [(required) = true];
        string network_load_balancer_name = 2 [(required) = true];
        yandex.cloud.loadbalancer.v1.Listener listener = 3 [(required) = true];
    }
}
